src/framework/components/text/system.js

```diff
@@ -0,0 +1,70 @@
+pc.extend(pc, function () {
+    /**
+     * @name pc.TextComponentSystem
+     * @description Create a new TextComponentSystem
+     * @class Attach 2D text to an entity
+     * @param {pc.Application} app The application
+     * @extends pc.ComponentSystem
+     */
+
+    var TextComponentSystem = function TextComponentSystem(app) {
+        this.id = 'text';
+        this.app = app;
+        app.systems.add(this.id, this);
+
+        this.ComponentType = pc.TextComponent;
+        this.DataType = pc.TextComponentData;
+
+        this.schema = [ 'enabled' ];
+
+        this.material = null;
+        this._shader = null;
+        this._defaultTexture = new pc.Texture(app.graphicsDevice, {width:4, height:4, format:pc.PIXELFORMAT_R8_G8_B8});
+    };
+    TextComponentSystem = pc.inherits(TextComponentSystem, pc.ComponentSystem);
+
+    pc.extend(TextComponentSystem.prototype, {
+        initializeComponentData: function (component, data, properties) {
+
+            if (data.text !== undefined) component.text = data.text;
```

Comment:
I'm using the component directly. 

Consider this an experiment in whether we should drop the `data` object altogether.